src/Model/Profile.php

```diff
@@ -263,8 +263,20 @@ private static function sidebar(App $a, array $profile, $block = 0, $show_connec
 		$o = '';
 		$location = false;
 
-		// This function can also use contact information in $profile
-		$is_contact = !empty($profile['cid']);
+		// This function can also use contact information in $profile, but the 'cid'
+		// value is going to be coming from 'owner-view', which means it's the wrong
+		// contact ID for the user viewing this page. Use 'nurl' to look up the
+		// correct contact table entry for the logged-in user.
+		$is_contact = !empty($profile['nurl']);
+		$contact = [];
+
+		if ($is_contact) {
```

Comment:
If `$is_contact` isn't used anywhere else, please replace it with the initial aassignment, in this case `!empty($profile['nurl'])`. The comment can be moved as well.

Reply:
Question is, if this variable does still make sense.

Reply:
`is_contact` is referred to when deciding whether or not to show the "Edit Profile" button. Line 359.

Reply:
This decision could be done with checking the "uid" value of the profile against the `local_user()` and to check whether "self" is set, I guess.